### ISSUE TITLE
Path error when running py_config from conda environment on Windows

### ISSUE THREAD

**Issue by langmm** [rstudio/reticulate#571]
After the most recent update, I am getting an error when running py_config from within a conda environment on a Windows appveyor CI test. The full output from my run (which are kind of hard to sort through) is [here](https://ci.appveyor.com/project/langmm/yggdrasil/builds/26462249), but the basic command was

```
Rscript --default-packages=methods,utils -e library(reticulate); reticulate::py_config()
```
and the output in R is

```
Warning message: package 'reticulate' was built under R version 3.6.1
Error in path.expand(path) : invalid 'path' argument
Calls: <Anonymous> ... conda_list -> unique -> normalizePath -> path.expand
```

r-base is installed from the r channel and reticulate is being installed from CRAN (into the conda environment).

My best guess is that the environments returned on line 60 of `conda_list` in reticulate/R/conda.R (`conda_envs <- fromJSON(conda_envs)$envs`) is not returning a valid path, but I am not very familiar with R and have had trouble getting much further in debugging. Also, I am not sure why conda installed r-base 3.6.0 rather than 3.6.1, but I wouldn't expect this to cause an issue with the returned path.

**Comment by langmm** [comment]
I have confirmed that using r-base 3.6.1 (from the conda-forge channel rather than r) yields the same result.

**Comment by kevinushey** [comment]
Unfortunately, I'm still not able to reproduce. Can you share the output of `reticulate:::conda_list()`? For example, I see:

```
> reticulate:::conda_list()
          name                                                           python
1    Anaconda3                     C:\\Users\\kevinushey\\Anaconda3\\python.exe
2    condatest    C:\\Users\\kevinushey\\Anaconda3\\envs\\condatest\\python.exe
3 r-reticulate C:\\Users\\kevinushey\\Anaconda3\\envs\\r-reticulate\\python.exe
```

**Comment by langmm** [comment]
The output I get is:

```
> reticulate:::conda_list():       
    name                                     python
    1 Miniconda3-x64             C:\\Miniconda3-x64\\python.exe
    2           test C:\\Miniconda3-x64\\envs\\test\\python.exe
```

(The `test` environment is the one that is active when the call is made if that matters)

I was able to avoid the error by explicitly setting the Python version to the conda Python using the `RETICULATE_PYTHON` environment variable. I'm not sure if it helps, but once I was able to do that I also ran 

`reticulage::py_versions_windows`:

<details>

```
> reticulate::py_versions_windows()    
     type hive   install_path            executable_path version arch
     1  PythonCore  HCU C:\\PY7FEF~1\\ C:\\PY7FEF~1\\\\python.exe     3.5  x64
     2  PythonCore  HCU C:\\Python35\\ C:\\Python35\\\\python.exe     3.5 i386
     3  PythonCore  HCU C:\\PY2ABB~1\\ C:\\PY2ABB~1\\\\python.exe     3.6  x64
     4  PythonCore  HCU C:\\Python36\\ C:\\Python36\\\\python.exe     3.6 i386
     5  PythonCore  HCU C:\\PYD487~1\\ C:\\PYD487~1\\\\python.exe     3.7  x64
     6  PythonCore  HCU C:\\Python37\\ C:\\Python37\\\\python.exe     3.7 i386
     7  PythonCore  HLM C:\\PYTHON~1\\ C:\\PYTHON~1\\\\python.exe     2.6  x64
     8  PythonCore  HLM C:\\PYTHON~2\\ C:\\PYTHON~2\\\\python.exe     2.7  x64
     9  PythonCore  HLM C:\\PYTHON~3\\ C:\\PYTHON~3\\\\python.exe     3.3  x64
     10 PythonCore  HLM C:\\PYTHON~4\\ C:\\PYTHON~4\\\\python.exe     3.4  x64
     11   Anaconda  HCU   C:\\MIB14C~1   C:\\MIB14C~1\\python.exe     3.5 i386
     12   Anaconda  HCU   C:\\MI8349~1   C:\\MI8349~1\\python.exe     3.5  x64
     13   Anaconda  HLM   C:\\MINICO~1   C:\\MINICO~1\\python.exe     2.7  x64
     14   Anaconda  HLM   C:\\MI8349~1   C:\\MI8349~1\\python.exe     3.5  x64
     15   Anaconda  HLM   C:\\MI8015~1   C:\\MI8015~1\\python.exe     3.6  x64
     16   Anaconda  HLM   C:\\MIE8D0~1   C:\\MIE8D0~1\\python.exe     3.7  x64
```
</details>

and `reticulate::py_config`:

<details>

```
> reticulate::py_config()              
     python:         c:\miniconda3-x64\envs\test\python.exe
     libpython:      c:/miniconda3-x64/envs/test/python27.dll
     pythonhome:     c:\MIC493~1\envs\test
     version:        2.7.16 |Anaconda, Inc.| (default, Mar 14 2019, 15:42:17) [MSC v.1500 64 bit (AMD64)]
     Architecture:   64bit
     numpy:          c:\MIC493~1\envs\test\lib\site-packages\numpy
     numpy_version:  1.16.4
     
     NOTE: Python version was forced by RETICULATE_PYTHON
```
</details>

**Comment by langmm** [comment]
Could this be because I am using Miniconda and not Anaconda?

**Comment by kevinushey** [comment]
It's possible, but hard to be sure.

You could also try checking to see what the output of:

```
conda <- reticulate:::conda_binary("auto")
system(paste(conda, "info --json")
```

looks like in your session, and check if any "strange" paths are being included in the `"envs"` key.

**Comment by jon-harmon-macmillan** [comment]
PR #576 (or I guess technically maybe something else that's in master right now) fixed this for me. I was running into it on Appveyor for https://github.com/jonathanbratt/RBERT/pull/13
Rebooting my personal machine also seems to have fixed it, although I didn't see *exactly* the same error locally.
I can't say for sure what caused it, but telling appveyor to install reticulate from github fixed it.

**Comment by khaeru** [comment]
I also encountered this bug using the combination of:
- Appveyor,
- Windows,
- Anaconda (specifically, Miniconda), and
- reticulate.

Big thanks to @langmm for suggesting the workaround of setting RETICULATE_PYTHON explicitly; this also worked for us. See the auto-linked PR above for details.

PS. before finding this issue I also found beringresearch/ivis#28 which appears to be the same error message, and perhaps the same root cause.

**Comment by ethanwhite** [comment]
Looks like I just re-reported this bug in #601, which was also with Miniconda. I tested it on three separate Windows 10 machines all with identical results. Updating to `master` also fixed this for me.

If possible I think a quick new release might be in order since this means the current release is broken for a reasonable fraction of Windows users.

**Comment by khaeru** [comment]
I was just able to remove, from iiasa/ixmp, my workaround for this bug; see [changes here](https://github.com/iiasa/ixmp/pull/267/files#diff-1cae281cb5a1ec297c0582be96e5c2f7L52). Our AppVeyor/Windows/Miniconda tests now pass, using reticulate **1.14**; see e.g. [this AppVeyor build log](https://ci.appveyor.com/project/iiasa/ixmp/builds/31093606/job/rmy00u9rhm2v74b0).

Since the 1.14 release contains the fix, and duplicate #601 is also closed, can this be viewed as fixed? I guess it is up to the reporter (@langmm) or the maintainer, depending on the practice they follow, to close the issue.

**Comment by langmm** [comment]
Sorry for taking so long, but yes with the release this is fixed. Thanks!
